### ISSUE TITLE
Add catalog serialize method

### ISSUE THREAD

**Issue by jhamman** [NCAR/intake-esm#166]
Some Intake drivers provide a `Catalog.serialize()` method that allows the user to cache their catalog. It would be useful if this was possible with intake-esm to allow users to document exactly which datasets were available at the time of an analysis.

```python
col = intake.open_esm_datastore(...)
col.search(experiment_id=['historical', 'ssp585'], source_id=models, table_id='Ofx', grid_label='gn')
cat_fx.serialize('yaml_or_json_file')
```

For reference, intake-dcat does this here: https://github.com/CityOfLosAngeles/intake-dcat/blob/master/intake_dcat/catalog.py#L74-L85

**Comment by andersy005** [comment]
Needs to be revisited once https://github.com/NCAR/esm-collection-spec/issues/13 is addressed.

**Comment by andersy005** [comment]
@bonnland implemented this a few months ago: https://github.com/NCAR/intake-esm/pull/195. It was released as part of v2020.03.16.

Thank you @jhamman & @bonnland! Closing...
